app/Http/Controllers/ChapterController.php

```diff
@@ -12,65 +12,72 @@
 class ChapterController extends Controller
 {
     public function find(int $book_id, $chapter_no)
-{
-    $book = Books::find($book_id);
-
-    if (!$book) {
-        return response()->json(['error' => 'Book not found'], 404);
-    }
-    $recent = null;
-    if (Auth::check()){
-    $user_id = Auth::user()->user_ID;
-
-    $recent = Recent::where('book_id', $book_id)
-        ->where('user_id', $user_id)
-        ->first();
+    {
+        $user = Auth::user();
+        $user_id = optional($user)->user_ID;
     
-    if ($recent === null) {
-        // The book is not in the recent list, so create a new entry with the chapter.
-        Recent::create([
-            'book_id' => $book_id,
-            'user_id' => $user_id,
-            'chapter' => $chapter_no,
-        ]);
-    } else {
-        // The book is already in the recent list.
-        if ($recent->chapter != $chapter_no) {
-            // Use the update method to update the chapter.
-            Recent::where('user_ID', $user_id)
-                ->where('book_id', $book_id) // Add this condition to restrict the update to the specific book
-                ->update(['chapter' => $chapter_no]);
+        // Update recent entry if user is logged in
+        if ($user_id) {
+            $recent = $user->recent()->where("user_id", $user_id)->first();
+    
+            if ($recent === null) {
+                Recent::create([
+                    'book_id' => $book_id,
+                    'user_id' => $user_id,
+                    'chapter' => $chapter_no,
+                ]);
+            } elseif ($recent->chapter != $chapter_no) {
+                Recent::where('user_id', $user_id)
+                    ->where('book_id', $book_id)
+                    ->update(['chapter' => $chapter_no]);
+            }
         }
-    }
-    }
-
     
-
+        $book = Books::find($book_id);
     
+        if (!$book) {
+            return response()->json(['error' => 'Book not found'], 404);
+        }
+    
+        $chapters = $book->chapters;
     
+        $chapter = $chapters
+            ->skip($user_id ? $chapter_no - 1 : 0)
+            ->take(1)
+            ->first();
     
+        if (!$chapter) {
+            return response()->json(['error' => 'Chapter not found'], 404);
+        }
     
+        // Calculate previous and next chapter numbers
+        $prev = ($chapter_no != 1) ? $chapter_no - 1 : -1;
+        $next = ($chapters->count() > $chapter_no) ? $chapter_no + 1 : -1;
```

Comment:
you can use the laravel standard for paginations - https://laravel.com/docs/10.x/pagination#converting-results-to-json